include/klee/util/Ref.h

```diff
@@ -139,6 +147,19 @@ class ref {
     return *this;
   }
 
+  // Move assignment operator
+  ref<T> &operator=(ref<T> &&r) noexcept {
+    using std::swap;
+    swap(ptr, r.ptr);
+    return *this;
+  }
+
+  // Move assignment operator
+  template <class U> ref<T> &operator=(ref<U> &&r) {
+    // swap
+    auto *tmp = cast_or_null<T>(r.ptr);
```

Comment:
Why use both `swap` and not swap?

Reply:
Here I need to typecast between different types, therefore `std::swap` is less straight forward to use.